### DISCUSSION TITLE
Custom Benthos builder

### DISCUSSION THREAD

**Discussion by sagikazarmark** [redpanda-data/connect#2329] [Ideas]
Caddy and k6 have their `xcaddy` and `xk6` custom builders that make it easier for people to build custom binaries with plugins embedded in them (without Go experience).

Do you think it would make sense for Benthos?

**Comment by mihaitodor** [comment]
Thanks for the suggestion @sagikazarmark! There are some prior suggestions in the same direction such as #2261. Eventually, it will be great to have something like that, yeah.

LE: Might be worth checking out the new [`wasm` processor](https://www.benthos.dev/docs/components/processors/wasm). If the custom plugin can be compiled to WebAssembly, then that's an easier way to go about it.
